code/game/objects/items/explosives/mine.dm

```diff
@@ -1,277 +1,1277 @@
+	light_range = radiation_range * 1.5
+	set_light(light_range, light_power, light_color)
+
+	var/list/exclusion_zone = circle_range(get_turf(src), radiation_range)	//Radiation SHALL NOT pass through walls (too much cope)
+	for(var/mob/living/carbon/radiation_victim in exclusion_zone)
+		if(!check_path(src, radiation_victim, FALSE, TRUE, TRUE, TRUE))	//circle_range() goes through walls, so make sure the target is reachable
+			exclusion_zone -= radiation_victim	//Remove it since it will be iterating again below
+			continue
+
+		//Apply initial damages of the detonation evenly in BURN and TOX, then do a fifth of it in cellular damage
+		radiation_victim.apply_damages(0, radiation_damage/2, radiation_damage/2, 0, radiation_damage/5, ishuman(radiation_victim) ? pick(GLOB.human_body_parts) : null, BIO)
+		radiation_victim.adjust_stagger(radiation_damage/5 SECONDS)
+		radiation_victim.adjust_radiation(radiation_damage SECONDS)
+
+	for(var/turf/irradiated_turf in exclusion_zone)
+		if(!check_path(src, irradiated_turf))
+			continue
+
+		for(var/obj/effect/temp_visual/radiation/rad_effect in irradiated_turf.contents)
+			qdel(rad_effect)	//Delete any present radiation effect, the new one will replace it
+
+		//Delete them before the next pulse otherwise the exclusion_zone list will be gigantic
+		new /obj/effect/temp_visual/radiation(irradiated_turf, time_between_pulses - 1)
+
+	if(!current_fuel)
+		deletion_timer = addtimer(CALLBACK(src, PROC_REF(disarm)), time_between_pulses - 1, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
+		return
+
+	pulse_timer = addtimer(CALLBACK(src, PROC_REF(extra_effects)), time_between_pulses, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
+
+/obj/item/mine/radiation/fueled/Initialize(mapload)
+	. = ..()
+	current_fuel = max_fuel
+
+/obj/effect/temp_visual/radiation
+	randomdir = FALSE
+	///How much damage each tick does
+	var/radiation_damage = 10
+	///Reference to the radiation particle effect
+	var/obj/effect/abstract/particle_holder/particle_holder
+
+/obj/effect/temp_visual/radiation/Initialize(mapload, effect_duration)
+	. = ..()
+	//Override the timerid value set on the object; the duration is determined by the radiation mine
+	deltimer(timerid)
+	timerid = QDEL_IN_STOPPABLE(src, effect_duration)
+	particle_holder = new(src, /particles/radiation)
+	var/static/list/connections = list(COMSIG_ATOM_ENTERED = PROC_REF(on_cross))
+	AddElement(/datum/element/connect_loc, connections)
+
+///Called when an atom enters the tile this radiation effect is on
+/obj/effect/temp_visual/radiation/proc/on_cross(datum/source, atom/A, oldloc, oldlocs)
+	if(!isliving(A))
+		return FALSE
+	if(irradiate(A))
+		START_PROCESSING(SSobj, src)
+
+/obj/effect/temp_visual/radiation/process()
+	if(!irradiate())
+		STOP_PROCESSING(SSobj, src)
+
+///Proc to run the radiation effects on mobs that are on the same tile as this radiation effect
+/obj/effect/temp_visual/radiation/proc/irradiate(mob/living/crosser)
```

Comment:
then you give them more aids constantly, which again has no dead check. For the love of god this needs to be nerfed beyond belief or removed from hvh because its completely rediculousl lol